packages/graphql/src/schema/augment/fulltext.ts

```diff
@@ -17,50 +17,114 @@
  * limitations under the License.
  */
 
-import { GraphQLFloat, GraphQLNonNull, GraphQLString } from "graphql";
+import type { GraphQLEnumValueConfigMap } from "graphql";
+import { GraphQLEnumType, GraphQLFloat, GraphQLNonNull, GraphQLString } from "graphql";
 import type { SchemaComposer } from "graphql-compose";
 import type { Node } from "../../classes";
 import { fulltextResolver } from "../resolvers/query/fulltext";
 import { upperFirst } from "../../utils/upper-first";
 import { FloatWhere } from "../../graphql/input-objects/FloatWhere";
 import { SCORE_FIELD } from "../../graphql/directives/fulltext";
 
-export const fulltextArgDeprecationMessage =
+export function createFulltextDeprecationMessage(node: Node, type = "argument") {
+    return `This ${type} has been deprecated and will be removed in version 4.0.0 of the library.
+Please use the fulltext argument in ${node.plural}Connection with index you wish to query instead.
+More information about the changes to @fulltext can be found here:
+https://neo4j.com/docs/graphql-manual/current/guides/v4-migration/#_fulltext_changes.
+`;
+}
+
+const fulltextArgDeprecationMessageInArgument =
     "This argument has been deprecated and will be removed in version 4.0.0 of the library. " +
-    "Please use the top-level query that corresponds to the index you wish to query instead. " +
+    "Please use the phrase and index arguments instead. " +
     "More information about the changes to @fulltext can be found here: " +
     "https://neo4j.com/docs/graphql-manual/current/guides/v4-migration/#_fulltext_changes.";
 
+function createEnumType(name: string, rawValues: string[]): GraphQLEnumType {
+    const values = rawValues.reduce((acc, value) => {
+        return {
+            ...acc,
+            [value]: {
+                value,
+            },
+        };
+    }, {} as GraphQLEnumValueConfigMap);
```

Comment:
```suggestion
    const values = rawValues.reduce((acc: GraphQLEnumValueConfigMap, value) => {
        return {
            ...acc,
            [value]: {
                value,
            },
        };
    }, {});
```

Just a little nicer not to use `as` in my opinion :)

Reply:
I tend to prefer strictly typing `reduce` functions, not having the `as` means that that reduce function now infers value to be an object (and an empty one at that!) as `reduce` uses the type of the second parameter to infer types

This means, that if I mess my code like:

```
    const values = rawValues.reduce((acc, value) => {
        return {
            ...acc,
            tst: "dsa",
            [value]: {
                value,
            },
        };
    }, {});
```

The typing system will not detect the problem, and furthermore, the returned `values` is still of type empty object so it gets silently passed without a problem.